scripts/samplesheet_generator.py

```diff
@@ -149,6 +146,57 @@ def gen_Hiseq_lane_data(pro):
 
     return ("{}{}".format(header, str_data), data)
 
+
+def gen_Novaseq_lane_data(pro):
+    data=[]
+    header_ar = ["FCID","Lane","Sample_ID","Sample_Name","Sample_Ref","index","index2","Description","Control","Recipe","Operator","Sample_Project"]
+    for out in pro.all_outputs():
+        if  out.type != "Analyte":
+            continue
+        for sample in out.samples:
+            sp_obj = {}
+            sp_obj['lane'] = out.location[1].split(':')[0].replace(',','')
+            sp_obj['sid'] = "Sample_{}".format(sample.name).replace(',','')
+            sp_obj['sn'] = sample.name.replace(',','')
+            try:
+                sp_obj['pj'] = sample.project.name.replace('.','__').replace(',','')
+            except:
+                #control samples have no project
+                continue
+            try:
+                if pro.udf.get('Read 2 Cycles'):
+                    if str(pro.udf['Read 2 Cycles']).replace(',','')==str(pro.udf['Read 1 Cycles']).replace(',',''):
+                        sp_obj['rc'] = "2x{}".format(str(pro.udf['Read 1 Cycles']).replace(',',''))
+                    else:
+                        sp_obj['rc'] = "{}-{}".format(str(pro.udf['Read 1 Cycles']).replace(',',''),str(pro.udf['Read 2 Cycles']).replace(',',''))
+                else:
+                    sp_obj['rc'] = "1x{}".format(str(pro.udf['Read 1 Cycles']).replace(',',''))
+            except:
+                sp_obj['rc'] = ''
+            sp_obj['ct'] = 'N'
+            sp_obj['op'] = pro.technician.name.replace(" ","_").replace(',','')
+            sp_obj['fc'] = out.location[0].name.replace(',','')
+            sp_obj['sw'] = out.location[1].replace(',','')
+            try:
+                sp_obj['ref'] = sample.project.udf['Reference genome'].replace(',','')
```

Comment:
You could write as `sp_obj['ref'] = sample.project.udf.get('Reference genome', '').replace(',','')` without `try` block